mettle/src/stdapi/sys/process.c

```diff
@@ -128,7 +128,33 @@ sys_process_get_processes(struct tlv_handler_ctx *ctx)
 struct tlv_packet *
 sys_process_attach(struct tlv_handler_ctx *ctx)
 {
-	return tlv_packet_response_result(ctx, TLV_RESULT_FAILURE);
+	bool inherit;
+	uint32_t pid;
+	struct tlv_packet *p = NULL;
+
+	tlv_packet_get_bool(ctx->req, TLV_TYPE_INHERIT, &inherit);
+	if(inherit)
+	{
+		return tlv_packet_response_result(ctx, TLV_RESULT_FAILURE);
+	}
+
+	tlv_packet_get_u32(ctx->req, TLV_TYPE_PID, &pid);
+	if(pid == 0)
+	{
+		uint32_t self_pid = getpid();
+		p = tlv_packet_response_result(ctx, TLV_RESULT_SUCCESS);
+		p = tlv_packet_add_u32(p, TLV_TYPE_PID, self_pid);
+		p = tlv_packet_add_u32(p, TLV_TYPE_HANDLE, self_pid);
+		return p;
+	}
+
+	p = tlv_packet_response_result(ctx, TLV_RESULT_SUCCESS);
+	p = tlv_packet_add_u32(p, TLV_TYPE_PID, pid);
```

Comment:
Let's also check the `TLV_TYPE_PROCESS_PERMS` permissions, and when a permission is requested that you can't support like create thread, all access, kill, etc, fail the operation with some kind of an error code since the current API can only perform memory reads.

Logging a reason that the operation failed as unsupported permission or something similar would be helpful too.